### ISSUE TITLE
Molotov Lit up Twice is death

### ISSUE THREAD

**Issue by philipp-gaddi** [CleverRaven/Cataclysm-DDA#61901]
### Describe the bug

When having a Molotov wielded and you activate it twice, it burns the 3x3 area around the player and the player. 

### Steps to reproduce

1. get a lighter and a molotov
2. wield the molotov
3. activate the molotov
4. activate the lit up molotov

### Expected behavior

i would expect a molotov be can't activated twice and burning the player.

### Versions and configuration

Linux 5.15.65-1-MANJARO
Game Version 26a4725dc07f8feedcc29cec91889d9140ddd5db
Ultica and CC Sound
English
No Mods

**Comment by RAldrich** [comment]
This appears to be a result of the Lit Molotov's: `"use_action": ["MOLOTOV_LIT"],` which does not specify any other parameters. I assume this is causing `iuse.cpp 3815 cata::optional<int> iuse::molotov_lit( Character *p, item *it, bool t, const tripoint &pos )` to pass t = false, which means the Molotov has no timer or fuse, and so it explodes in your hand.

tldr; yeah, seems like a bug. If there was a timer or fuse that had not burned out yet, you'd get a message about it already being lit, or the fuse burning out harmlessly.

**Comment by RAldrich** [comment]
Ah ha! Some digging indicates that this code is meant to prevent you from 'd'ropping the Molotov. They're meant to blow up if you do that. 'a'ctivating it a second time is likely making it think you're no longer holding it, and so it blows up.

**Comment by philipp-gaddi** [comment]
interesting, the measure to prevent it caused it in the end :D, great that you found out how to fix it, can i close this yet?

**Comment by RAldrich** [comment]
Go ahead and leave this issue open until the PR is actually merged. It should auto-resolve it then.
